chpl/chpl-service/src/main/java/gov/healthit/chpl/service/CertificationCriterionService.java

```diff
@@ -39,14 +40,17 @@ public CertificationCriterionService(CertificationCriterionDAO certificationCrit
 
     @PostConstruct
     public void postConstruct() {
-        criteriaMap = certificationCriterionDAO.findAll().stream()
+        criteriaByIdMap = certificationCriterionDAO.findAll().stream()
                 .map(criterion -> new CertificationCriterion(criterion))
                 .collect(Collectors.toMap(CertificationCriterion::getId, cc -> cc));
+        criteriaByNumberMap = criteriaByIdMap.values().stream()
+                .collect(Collectors.groupingBy(CertificationCriterion::getNumber));
+
         referenceSortingCriteriaList = getReferenceSortingCriteriaList();
     }
 
     public CertificationCriterion get(Long certificationCriterionId) {
-        return criteriaMap.containsKey(certificationCriterionId) ? criteriaMap.get(certificationCriterionId) : null;
+        return criteriaByIdMap.containsKey(certificationCriterionId) ? criteriaByIdMap.get(certificationCriterionId) : null;
```

Comment:
Can this be just the `criteriaByIdMap.get(certificationCriterionId)` call? As that call returns null if the key doesn't exist?